From 4852d1c0439a98b4ed112a33e490de94daad12f5 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sat, 1 May 2021 01:28:07 +0200
Subject: [PATCH] input: goodix: Don't disable regulators during suspend

It does no harm to disable them, but on Pinephone we have other
peripherals attached to the same power supplies, whose drivers
will not reference the regulator, so powering down the regulators
from Goodix driver will break those other devices.

Until those drivers gain the regulator support, don't disable
the regulators in Goodix driver.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/input/touchscreen/goodix.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index d1a9d97e4..796763678 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1278,8 +1278,6 @@ static int __maybe_unused goodix_suspend(struct device *dev)
 		goodix_free_irq(ts);
 		goodix_irq_direction_output(ts, 0);
 		gpiod_direction_output(ts->gpiod_rst, 0);
-		regulator_disable(ts->avdd28);
-		regulator_disable(ts->vddio);
 		return 0;
 	}
 
@@ -1327,18 +1325,6 @@ static int __maybe_unused goodix_resume(struct device *dev)
 	int error;
 
 	if (ts->poweroff_in_suspend) {
-		error = regulator_enable(ts->avdd28);
-		if (error) {
-			dev_err(dev, "Regulator avdd28 enable failed.\n");
-			return error;
-		}
-
-		error = regulator_enable(ts->vddio);
-		if (error) {
-			dev_err(dev, "Regulator vddio enable failed.\n");
-			return error;
-		}
-
 		error = goodix_reset(ts);
 		if (error) {
 			dev_err(dev, "Controller reset failed.\n");
-- 
2.37.2

